### PR TITLE
TYP: type up parts of series.py

### PR DESCRIPTION
**Pull request by topper-123**
More typing.

### CODE REVIEW

**Review by jreback — APPROVED** [review]

File: pandas/core/generic.py [hunk 1]
```diff
@@ -8866,9 +8865,7 @@ def mask(
     """
 
     @Appender(_shared_docs["shift"] % _shared_doc_kwargs)
-    def shift(
-        self: FrameOrSeries, periods=1, freq=None, axis=0, fill_value=None
-    ) -> FrameOrSeries:
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
```
Comment (WillAyd) [inline pandas/core/generic.py]: Why was this removed? I think this is correct as is?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py [hunk 2]
```diff
@@ -4099,7 +4099,6 @@ def add_suffix(self: FrameOrSeries, suffix: str) -> FrameOrSeries:
 
     def sort_values(
         self,
-        by=None,
```
Comment (simonjayhawkins) [inline pandas/core/generic.py]: DataFrame.sort_values is not typed. This signature will not now be compatible?

Comment (simonjayhawkins) [reply pandas/core/generic.py]: mypy now gives (when DataFrame.sort_values is typed)
pandas\core\frame.py:4867: error: Signature of "sort_values" incompatible with supertype "NDFrame"
instead of 
pandas\core\series.py:2776: error: Signature of "sort_values" incompatible with supertype "NDFrame"

so deleting this is not solving the issue.

File: pandas/core/generic.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/generic.py]: There's 3 possible returns here.
self.copy()
self.tshift(periods, freq)
self._constructor(new_data).__finalize__(self)

self.copy() has only 1 possible return.
self._constructor(data).__finalize__(self)

self.tshift has 2 possible returns:
self
self._constructor(new_data).__finalize__(self)

making the assumption that self._constructor is restricted to the type of self (maybe should discuss), then this is correct as is.

Series.shift may need to be changed to a typevar at some point so that subclassed Series returns a subclassed Series, but I'm not seeing any mypy errors if this is left as is, so not sure why this is changed?

File: pandas/core/generic.py (hunk 2 above)
Comment (WillAyd) [reply pandas/core/generic.py]: Note there is a dedicated issue #30451 for sort_values; would be OK with tackling that on its own as I think will be tricky

**Comment by topper-123** [comment]
I think this is the best I can do here: I agree a TypeVar would be best, but it doesn't play well with ``._constructor`` AFAICS.

I wonder if we should deprecate ``._constructor`` and use ``type(self)`` for instantiation, but that's probably a different discussion wit its own traps.

File: pandas/core/generic.py (hunk 2 above)
Comment (WillAyd) [reply pandas/core/generic.py]: @topper-123 can you restored this deleted annotation? IIUC from @simonjayhawkins above doesn't need to be removed

Otherwise lgtm happy to get this merged

**Comment by jreback** [comment]
lgtm.

File: pandas/core/generic.py (hunk 1 above)
Comment (WillAyd) [reply pandas/core/generic.py]: @topper-123 sorry I commented on the wrong one before. Can this not be restored per @simonjayhawkins comment above?

Comment (topper-123) [reply pandas/core/generic.py]: Yeah, looks like you're right. I've pushed a new commit.

**Review by WillAyd — APPROVED** [review]
lgtm @simonjayhawkins

**Comment by topper-123** [comment]
ping

File: pandas/core/series.py [hunk 3]
```diff
@@ -1425,7 +1439,7 @@ def to_markdown(
 
     # ----------------------------------------------------------------------
 
-    def items(self):
+    def items(self) -> Iterable[Tuple[Optional[Hashable], Any]]:
```
Comment (simonjayhawkins) [inline pandas/core/series.py]: @WillAyd added _typing.Label in #29140

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@topper-123 lgtm ex comment. ping on green.

**Comment by simonjayhawkins** [comment]
Thanks @topper-123
